subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+				found = true
+				policy, _, err := consul.ACL().PolicyRead(p.ID, nil)
+				require.NoError(err)
+				if expected == "connect-inject-token" {
+					// The connect inject token doesn't have namespace config,
+					// but does change to operator:write from an empty string.
+					require.Contains(policy.Rules, "operator = \"write\"")
+				} else {
+					require.Contains(policy.Rules, "namespace")
+				}
+				break
+			}
+		}
+		require.True(found, "Did not find "+expected)
+	}
+}
+
+// Test the auth method and acl binding rule created when namespaces are enabled
+// and there's a single consul destination namespace.
+func TestRun_ConnectInject_SingleDestinationNamespace(t *testing.T) {
+	t.Parallel()
+	k8s, testAgent := completeEnterpriseSetup(t, resourcePrefix)
+	defer testAgent.Stop()
+	setupK8sServiceAccount(t, k8s)
+	require := require.New(t)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	args := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-create-inject-auth-method",
+		"-enable-namespaces",
+		"-consul-inject-destination-namespace=dest",
+	}
+
+	responseCode := cmd.Run(args)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	bootToken := getBootToken(t, k8s, resourcePrefix)
+	consul, err := api.NewClient(&api.Config{
+		Address: testAgent.HTTPAddr,
+		Token:   bootToken,
+	})
+	require.NoError(err)
+
+	// Check the ACL auth method is created in the expected namespace.
+	authMethodName := "release-name-consul-k8s-auth-method"
+	methods, _, err := consul.ACL().AuthMethodList(&api.QueryOptions{
+		Namespace: "dest",
+	})
+	require.NoError(err)
+	found := false
+	for _, m := range methods {
+		if m.Name == authMethodName {
+			found = true
+		}
+	}
+	require.True(found, fmt.Sprintf("no auth method with name %s found", authMethodName))
+
+	// Check the binding rule is as expected.
+	rules, _, err := consul.ACL().BindingRuleList(authMethodName, &api.QueryOptions{
+		Namespace: "dest",
+	})
+	require.NoError(err)
+	require.Len(rules, 1)
+}
+
+// Test the auth method and acl binding rule created when namespaces are enabled
+// and we're mirroring namespaces.
+func TestRun_ConnectInject_NamespaceMirroring(t *testing.T) {
+	t.Parallel()
+	k8s, testAgent := completeEnterpriseSetup(t, resourcePrefix)
+	defer testAgent.Stop()
+	setupK8sServiceAccount(t, k8s)
+	require := require.New(t)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	args := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-create-inject-auth-method",
+		"-enable-namespaces",
+		"-enable-inject-k8s-namespace-mirroring",
+		"-inject-k8s-namespace-mirroring-prefix=prefix-",
```

Comment:
There should be another test case with an empty prefix. Also, it would be good to test that providing a `-consul-inject-destination-namespace` field gets overridden by the mirroring.

Reply:
Added cases:
```go
	cases := map[string]struct {
		MirroringPrefix string
		ExtraFlags      []string
	}{
		"no prefix": {
			MirroringPrefix: "",
			ExtraFlags:      nil,
		},
		"with prefix": {
			MirroringPrefix: "prefix-",
			ExtraFlags:      nil,
		},
		"with destination namespace flag": {
			MirroringPrefix: "",
			// Mirroring takes precedence over this flag so it should have no
			// effect.
			ExtraFlags: []string{"-consul-inject-destination-namespace=dest"},
		},
	}

```